src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -60,6 +60,23 @@ static std::vector<std::shared_ptr<ov::Node>> cast_to_node_vector(const ov::Sink
     return nodes;
 }
 
+// Assign operations created via Assign py binding have Variables which are not connected to
+// ReadValue operations. This function attempts to resolve this situation by finding correct Variables
+// for Assigns.
```

Comment:
why is it done on python bindings level? is it python specific?

Reply:
Assign/ReadValue pair has to be initialized with the same obj of Variable class, this a requirement.
The previous solution is a` map <string, shared_ptr<Variable>> ` inside NodeFactory class. This map allows to get Variable by name, if it exists, or stores a created Variable, if it doesn't. So it helps us to use the same Variable obj for Assign/ReadValue classes.
But a separate py binding was created for Assign class couple Releases ago, it means that we use the class constructor to create Assign operation instead of NodeFactory. It broke the logic above and now it's impossible to create a correct ov::Model with Assign/ReadValue via py api.

Reply:
It was decided to use Model class binding because Assign binding doesn't have access to the model and we can' t get all Variables on this level.